src/main/java/Duke.java

```diff
@@ -1,10 +1,110 @@
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
+        String name = "Duke";
+        String input;
+        Scanner reader = new Scanner(System.in);
+
+        String logo = " ____        _\n"
+                + "|  _ \\ _   _| | _____\n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println("\tHello from\n" + logo);
+
+        printLine();
+        welcome(name);
+        printLine();
+
+        do {
+            input = reader.nextLine();
+            printLine();
+            if (input.equals("bye")) {
+                // close the interpreter
+                System.out.println("\tBye. Hope to see you again soon!");
+            } else if (input.equals("list")) {
+                // list all tasks
+                listTasks();
+            } else {
+                // input involves 2 phrases
+                int startIndexOfInput = input.indexOf(' ');
+                String cmdType = input.substring(0, startIndexOfInput).toLowerCase();
+                String cmdDescription = input.substring(startIndexOfInput + 1);
+
+                if (cmdType.equals("done")) {
+                    // mark a task as done
+                    System.out.println("\tNice! I've marked this task as done:");
+                    int indexOfTasks = Integer.parseInt(cmdDescription) - 1;
+                    tasks[indexOfTasks].setDone(true);
+                    System.out.println("\t" + tasks[indexOfTasks]);
+                } else {
+                    // add stuff to String[] tasks
+                    addTask(cmdType, cmdDescription);
+                }
+            }
+            printLine();
+        } while (!input.equals("bye"));
+    }
+
+    public static void printLine() {
+        System.out.println("\t____________________________________________________________");
+    }
+
+    public static void welcome(String name) {
+        System.out.println("\tHello! I'm " + name);
+        System.out.println("\tWhat can I do for you?");
+    }
+
+    public static void addTask(String type, String description) {
+        if (indexOfTasks >= MAX_TASKS_COUNT) {
+            // exit if task count has exceeded
+            return;
+        }
+
+        if (type.equals("todo")) {
+            tasks[indexOfTasks] = new ToDo(description);
+        } else if (type.equals("deadline")) {
+            String[] cmdArgs = new String[2];
+            cmdArgs = processArgs(description);
+            tasks[indexOfTasks] = new Deadline(cmdArgs[0], cmdArgs[1]);
+        } else if (type.equals("event")) {
+            String[] cmdArgs = new String[2];
+            cmdArgs = processArgs(description);
+            tasks[indexOfTasks] = new Event(cmdArgs[0], cmdArgs[1]);
+        }
+
+        Task.incrementTaskCount();
```

Comment:
Interesting idea, instead of putting the counter in the main 🥇

Reply:
This has the assumption that there is only 1 Task object instantiated at anytime, so it might be better to record taskCount in class Duke?